pandas/_libs/tslib.pyx

```diff
@@ -523,15 +523,14 @@ cpdef array_to_datetime(
 
             elif is_integer_object(val) or is_float_object(val):
                 # these must be ns unit by-definition
-                item_reso = NPY_FR_ns
-                state.update_creso(item_reso)
-                if infer_reso:
-                    creso = state.creso
 
                 if val != val or val == NPY_NAT:
                     iresult[i] = NPY_NAT
```

Comment:
So we no longer need to update the creso state here?

Reply:
we delay until after the `if val != val or val == NPY_NAT` so as to treat these cases symmetrically with other NaTs